app/assets/src/components/views/amr_heatmap/AMRHeatmapControls.jsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Divider } from "~/components/layout";
+import { Dropdown } from "~ui/controls/dropdowns";
+
+import cs from "./amr_heatmap_view.scss";
+
+export default class AMRHeatmapControls extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  generateChangeFunction(filter) {
+    const changeFunction = option => {
+      if (option === this.props.selectedOptions[filter]) {
```

Comment:
we usually have a convention of pulling out all the props and state at the top of the function like

`const { selectedOptions, onSelectedOptionsChange } = this.props;`

Reply:
you could probably just replace this function with something like

`() => option !== selectedOptions[filter] && onSelectedOptionsChange({ [filter]: option })`

Reply:
I pulled out the props at the top of the function, but for the sake of readability I kept the control flow clearer (but shortened it). Because of the way the Dropdown component handles the `onChange` callback function, there's no way to access `filter` in the callback, which is why I decided to generate it beforehand.